### ISSUE TITLE
Docedit Info Refined

### ISSUE THREAD

**Issue by FelesNoctis** [wrye-bash/wrye-bash#432]
**Active:**
`feles-docedit-grndbeef` _(current dev-appropriate branch)_
`feles-docedit-postcred` _(don't use, experimental)_

**Rebased, Outdated:**
`Wrye-Bash-Collection: feles-docedit-rebased-live`

**Dead:**
`feles-docedit`
`feles-docedit-live`
`feles-docedit-sandbox`

**Status:**
`feles-docedit-grndbeef` is content complete, should be ready for merge.
`feles-docedit-postcred` has DO NOT CROSS tape everywhere, don't touch!

---

**Currently Implemented `-postcred`:**
_All items here are tentative and subject to change._
- Sticky headers have been applied to with multiple redundancies.
-- Javascript used for to-pixel sticky measurements, with CSS using approximate, both within support-only arguments.
- Section tags have been added to enable Sticky qualities. These are complex, see upcoming section.

**Upcoming Experiments `-postcred`:**
- Move navigation menus to fixed sidebar, with folding and off-screen+tab hiding when unneeded.
- Re-implement smooth scroll in a cross-compatible and consistent way (scroll-behavior CSS is limited to Firefox and Chrome)
- Custom tags for Sticky functionality may be moved to javascript as DOM-manipulations instead for ease of future editing of this readme generation.

**Comment by FelesNoctis** [comment]
I also apologize for the documentation time expectation not being met. I spent the first half of that night trying to figure out why git was "broken" (see #260), and then the other half setting up my own fork so I could continue, thinking I'd been kicked out of the club! Monday and today weren't very good for time, but I did get a couple of things done. Tomorrow (Wednesday) afternoon is currently mostly open, which unless everything goes horribly south again should be more than enough time to finish what I'd planned for up to the Tentative steps.

**Comment by Utumno** [comment]
Thanks @FelesNoctis

**Comment by FelesNoctis** [comment]
As much as I hate to admit it, that's another missed expectation gone. Turns out what I figured would be an easy task turned into something that dragged on until almost 6am, and apparently trying to float a semi-broken slideshow JS when my brain isn't functioning means I now have this:
![Splat](https://giant.gfycat.com/LinearThankfulAdamsstaghornedbeetle.gif)
Everything was going so well, too.

So. Re-prioritizing. That float idea? I'm tossing it. That's what's caused a good 90% of my issues in the first place between fighting with the javascript, trying to correctly position the caption without margins, and has generally just been a royal pain in the rear end since day one.

Screw it. I plan on starting on the overhaul as soon as I finish patching these ones together anyway. No sense wasting more of my time.

TODO, 2019-01-24:
- Hope to whatever god that I can drag myself out of the coma that I just caused myself at a decent hour
- Make breakfast/lunch/afternoon snack?
- Laugh as everything causing my float-related fury is shredded and sails neatly into the wind
- Suck up my pride, save my best efforts for the future, and just nail the most basic changes out, like I _freaking volunteered myself for in the first place_, and worry about fanciness for the next iteration.

Now if you'll excuse me, I'm going to go grumble my way to bed

**Comment by Utumno** [comment]
Ok no worries - but yes let's have the basic one cause we need to edit the readme itself a bit :P

**Comment by FelesNoctis** [comment]
@Utumno 
It's now past 5am, and I need to be up in a little over an hour. I should have been done a long time ago but when I told my local repo to switch gits, it "crashed" and looked like it deleted half my stuff. Really, as in the git thread itself crashed. Thankfully I discovered it had been automatically stashed, but... heart attacks in the early AM aren't a good thing.

Also I screamed in terror when I clicked the Push button because I felt like it was appropriate.
_Don't judge me._

The Readmes should be in a place where they're editable for the new stuff. Notes for styling the floats and all that is in **bash_main.css** at the top. You could also just have things added as you normally would and I'll come through after and clean it up. Either way works.

Also you know the images you wanted me to do originally? There's more. When I added the ones we DID have to the appropriate places, I realized there were even more context menus that hadn't been created at all. So uh... yeah. Guess I'll be taking more screenshots. :P

**Comment by FelesNoctis** [comment]
@Utumno Can I get a list of current/recent contributors and their roles so the credits section can be updated?

**Comment by Utumno** [comment]
Well here is some https://github.com/wrye-bash/wrye-bash/graphs/contributors although the stats are a bit inexact, plus apart from lojack, major contributors like err Wrye, Warudar, haama and I may be forgetting someone are missing (passificMorrowind is Nick Wilde) - I would like to merge so do you think we are ready (first pass that is)?

**Comment by FelesNoctis** [comment]
The current state is just fine for a merge. What you wanted me to update originally is done, the files were reformatted and should be HTML5 compliant (beyond minor inline styles), and the images have been optimized (but are still in their original places).

I have more to do this afternoon but that can always be wrapped in at a later time. The important stuff is done.

**Comment by FelesNoctis** [comment]
Ayyyy!
***fingerguns***
I got myself moved over to the other branch AND got my changes on both, and it only took... 4 tries and 20 minutes! That's an improvement! I'm so proud of myself.

Next is getting the version history formatted, as well as rechecking to see what screenshots still need to be made for the advanced readme. I know there's some but I forgot which. I think they're context menus. Oh, and re-designing the credits area, and possibly turning the nav menus into a fixed position block to the left instead of only being at the top, for easier navigation.

Gettin' all fancy up in here.

**Comment by Utumno** [comment]
So what are we doing here @wrye-bash/bashers ?

**Comment by Utumno** [comment]
Unfortunately there are still issues:

![clipboard01](https://user-images.githubusercontent.com/1365079/52520769-03060500-2c76-11e9-96b1-934ead9523e1.jpg)
![clipboard02](https://user-images.githubusercontent.com/1365079/52520770-039e9b80-2c76-11e9-8720-060e2babee10.jpg)

@FelesNoctis you changed the formatting of the readmes which makes the whole file a huge diff so I can't really review. It would be better if you had avoided formatting the readme and just changed the lines that needed to change. Now I am unsure what changes I will be really merging to dev and this makes me reluctant to merge

**Comment by FelesNoctis** [comment]
Interesting, the code block isn't breaking on my version or at least wasn't last I checked, I'll double-check that as soon as I get home in... 9 hours or so.

In the second screenshot, what resolution are you running at? The changes were designed for 16x9, and I'm guessing you're running 9x16 judging by the portrait layout? I didn't test that aspect ratio (my testing would have been at 8x9 now that I think about it, where it looks normal and pushes the text under), so I'll have to take another look.

As for what you're "merging", my comment in the beta thread. This isn't a merge, it's a **replacement**. The old documentation had errors in the hundreds to thousands depending on the strictness of the validation, including unclosed tags everywhere. I mentioned this back on discord when I first started on it, and that I did a full formatting pass. If you're concerned about how it matches to dev, give me tonight to run a comparison and I'll make sure there isn't anything sneaky left behind.

**Comment by FelesNoctis** [comment]
@code block format breakage
Again I'll take a look when I get back, but this is an older version from a week ago or so on my fork: looks just fine

https://felesnoctis.github.io/wrye-bash-docedit/Mopy/Docs/Wrye%20Bash%20General%20Readme.html#trouble-known

**Comment by FelesNoctis** [comment]
@code block format breakage
Again I'll take a look when I get back, but this is an older version from a week ago or so on my fork: looks just fine on my phone. Or at least as good as it can given a vertical layout on a mobile device. :P No breakage here.

https://felesnoctis.github.io/wrye-bash-docedit/Mopy/Docs/Wrye%20Bash%20General%20Readme.html#trouble-known

**Comment by FelesNoctis** [comment]
@Utumno
Three messages! Sorry, thought scattered while getting ready for work. Summarizing from above:

- Please only work with **felis-docedit-rebased**. Not sure if any of the others exist (hard to tell on the phone), but only that one should be used.
- My fork that was linked above does not have the code block issue, and last I saw neither does my local version. Please tell me what browser and OS you're using so I can check it. I have 4 on my comp and 3 on my phone, and chrome can emulate, so I can cover quite a bit.
- The image wrap issue is unique to tall a
aspect ratios. I tested at half-standard (8x9), but apparently forgot portrait (I'm assuming you're 9x16). My fault 100%. My plan for white space handling had a backup: centering the images. For the sake of brevity I'll fall back to that instead of fiddling more.
- Because this is a **replacement** and not a **merge**, I can do the merging of the docs manually on my end first. Please let me know which branches you're merging and I'll handle doc comparison and merging on my end so you can just send mine in wholesale.
- My original plan was to get this all done tonight after work for you. However as I was walking out the door for work, the power went out. Pray for me, hopefully it'll be back on by the time I get home.

**Comment by FelesNoctis** [comment]
An update, it's almost 9pm, still no power, and the last update was 530pm with no ETA. Here's hoping the morning brings better news.

In the meantime:

Codeblock stretch is due to "pre"formatted text no longer being formatted. Thank you for pointing that out because it does need to be fixed, but what you're seeing is it of date. I'd removed the pre tag a while ago.

Again, please let me know what branches are intended to be merged so I can edit the relevant changes in by hand.

**Comment by Utumno** [comment]
@FelesNoctis I need the thing redone so it only has the needed changes in. The branch we agreed upon I the rebased one but I would like you to create a new branch over `dev`  and do a series of small commits applying the changes. So one commit closing the tags and doing just this. One commit moving the css to files. One commit rewritting the images. One commit changing the credits section and adding the pics you need there. No changes in formatting like putting the li items in different lines. This way I can review and be sure that nothing changes in the readme content.

As for git: please _only_ use git for windows. As it is clear the tools you and other people use result in a chaos. The workflow is you launch `gitk --all &` and `git gui`, you checkout your branch you make small changes you commit them _in the gui_ you check the tree and then you push _from the git gui_. I can help with git for windows

**Comment by FelesNoctis** [comment]
@Utumno
The error in question regarding the code block: The issue stems from the undoing of the merging you had to spend the day doing when an unnamed someone left the team, as evidenced by the fact that my fork during that time has the correct changes. I expected the reversion here would be the same, and I was wrong, but it was such a small thing it went unnoticed. Why my local version still doesn't show it I don't know.

The "knot" between live and rebased: That was due to my own stupidity, and not any extra tools as you claim. The tool that I used during that time, I can easily open the log and see exactly what it does: running the commands directly through the CLI exactly as you'd expect. No additional steps, no extra frills. The error came in my attempt to force one local repo to push to two places. I realized after that instead of doing that, I should have pulled -rebased and then just cherry-picked off of -live and pushed that instead. Given the confusion I tend to have with git in the first place, as I've mentioned to you a number of times, *just using Git for Windows would make things worse, not better.* It confuses the hell out of me. I'm actually learning more and faster with these other tools than if I just continued to fumble with GfW.



Now, if I can be blunt with you for a moment.

This messages left at 4am on random days with otherwise no other contact isn't conducive to a well-functioning project. Many of the changes I made I informed you of (such as the formatting overhaul) and you gave your approval, or at least we're indifferent. Now that the work itself is long since done you're suddenly deciding it ISN'T ok and not only telling me to start over, but to take steps that would require excessive amounts of manual labor over most of the day, if not longer, to integrate by hand.

There was more I was going to add here, but decided it should probably be kept out of a public forum. I would like to speak with you privately in a real-time chat at your earliest convenience. If scheduling is an issue, let me know a good time for you and I can make sure I'm around then

**Comment by Utumno** [comment]
Sorry @FelesNoctis if I sounded harsh, I have very limited time and apparently, the main line of development is much more time consuming than I thought.

I am not a fan of real time chats will do if I have any time but don't hold your breath :P

I am not asking to redo rather to carefully go through and separate commits are away of doing so

**Comment by FelesNoctis** [comment]
Unfortunately, due to the need to remove the W3 standards compliance so that you can properly review as you're asking, I DO need to redo from step 0.
![Compliance Conflicts](https://i.imgur.com/nKthKBz.png)
That's the conflicts as they currently stand, which you likely already know. The standards pass was done over a month ago, with all of my work to date building upon it. If I were to remove that in order for you to be able to review as you're expecting, it would require starting over. There's no way to use later commits to check for conflicts for those steps either, because each one would still cover 90% of the file in conflicts. I could check changes manually between newer versions on the site, but between needing to cross-reference multiple files, ensuring that what I'm carrying over changes as they're intended to be and not breaking layout, and the other steps necessary to do so, it may actually take more time than just starting over.

**Comment by FelesNoctis** [comment]
And I apologize if this is coming across as aggressive. I've been talking this over with multiple people, since I read this, and I'm feeling like nobody is really full comprehending what's required here.

**Comment by FelesNoctis** [comment]
Before Utumno gets to say anything about this, I just wanna show this:

![knotting again](https://i.imgur.com/VeKlCxv.png)

I amended to add a file, sent it forward, and suddenly it was saying that I diverged. I pulled, and it told me that I was already up to date.

So once again, Git can bite me :P

**Comment by Utumno** [comment]
This should not happen - or rather it happens because you did something before you shouldn't have done. Solution is _never ever_ git pull - this is what led to the merges we saw. The remote branch in this case you must always force push. To make sure nothing goes wrong you should `git fetch` and then launch `gitk --all &` to inspect what is going on.

**Comment by FelesNoctis** [comment]
> Solution is _never ever_ git pull - this is what led to the merges we saw.
That's something Ganda was just teaching me about. A lot of the information I found that tries to "teach" git seems to imply that pull is what you're supposed to do. Unfortunately, it seems like a lot of the lessons on git are written by people who expect you to know how to use git in order to learn how to use git. **facepalm**

The `git fetch` was done last night before my next two pushes, and the mess was already inspected. Like last time it's just a knot in my end that doesn't appear to have had any significant effect on the branch itself, other than something that looks like a mess. It's ignorable from what I can tell.

I'll keep _force push_ in mind for the future. This started after I stashed changes that I can't imagine would have caused the supposed desync with the remote, but apparently did somehow. Considering the local files shouldn't have been behind in any way from the remote, I'm just going to keep the force push option in my toolbelt, finish reading what Ganda sent me, and move on.

**Comment by FelesNoctis** [comment]
And for your sanity, following what Sharlikran had helped me with the other day, I just did all this. Hopefully between force push and this I won't have to worry about chaos when trying to update both WB's main and my fork.

https://i.imgur.com/oEVzhFI.png
https://i.imgur.com/wH2nKWw.png

And all from the terminal for your sanity. :P

**Comment by Utumno** [comment]
I seldom use the terminal compared to how often I use the tree (gitk) and gui. I never (almost) use git status. I go to the gui and refresh so I not only see what changed but also how. Adding another remote (your fork) is fine and dandy but you don't really need to push there that's why I gave you write rights here.

Will have a look to the branch ASAP

**Comment by Utumno** [comment]
I'm impressed - that's nearing reviewable :P

Still confused by the middle of your branch - can you replace all those merges with a single commit:

![clipboard01](https://user-images.githubusercontent.com/1365079/52635711-4b6e2e80-2ed3-11e9-8250-b5891cc8b68e.jpg)

I will let you into a little trick: https://stackoverflow.com/a/17549542/281545

**Comment by GandaG** [comment]
@Utumno what git mergetool do you use? I've grown to dislike vimdiff but
other gui tools aren't as easy to work with.

**Comment by Utumno** [comment]
I do it manually -_-

**Comment by Utumno** [comment]
(but have a look here at the git tips wiki page for a script to keep only yours/theirs)

**Comment by Utumno** [comment]
The right way to resolve conflicts is having very few - in other words write interchangeable commits - that's the art :P

**Comment by FelesNoctis** [comment]
@Utumno 
> Still confused by the middle of your branch - can you replace all those merges with a single commit

All of that chaos from the first "Protection" to the last merge "Headers" commit should have been a single commit. That's when I stashed some changes in order to amend Protection and things went haywire. Those merges should be blank, I believe I checked them before. There's no reason for them to exist beyond my dunce hat. I'll be perfectly honest, I'd rather one of you handled the chaos in there. Regardless of what info I'm given, I'm still paranoid. :P I'd rather leave it to the people who know more.

> I'm impressed - that's nearing reviewable :P

NEARING?! I'll have you know those are the smallest commits I've done to date! They damn well better be reviewable! 🙄 

Also, for your sake, I've been leaving the tag closing step until the very last. There's no saving the "review" step from that one, it's going to cover the files in somewhere between 50-75% conflicts. I figure I can get the actual "meat" (*cough*) of the work done so you can just look at that step, cry a little, and move on.

As for the names and the commit comments... that's my revenge for having to do this again. 😈

**Comment by FelesNoctis** [comment]
Oh, and I forgot about the fork remote.

> Adding another remote (your fork) is fine and dandy but you don't really need to push there that's why I gave you write rights here.

I've also experienced how suddenly that can be removed. Call it a safety net, that way I can just swap to the other repo if something were to happen again. :P 

Also I intend on pushing a lot of the full overhaul work to my own fork first anyway, since it's such a massive project starting from step 0 there's no git-review process possible. Someone's going to have to just look it over as it stands.

**Comment by Utumno** [comment]
> **I'd rather one of you handled the chaos in there**.

Will do _in person_ mwhahaha

> They damn well better be reviewable!

Haha ok they are - pressed for time will get there in next couple days

> Also, for your sake, I've been leaving the tag closing step until the very last.

Excellent!

> Call it a safety net,

Or a sandbox for understanding git. The key to git is gitk - pan intended, but mark my words

So now let me concentrate ooooommmmm and out

**Comment by FelesNoctis** [comment]
So I've been delaying the content additions for a bit (mostly for my own sanity, picking through the previous branch is a bit tedious. Shhhh) and poking at the slideshow again.

There were complications with the floating previously due to how the slideshow and the static images function differently, and that carried over to the centering and scaling I did here. I DID get it working in a way that works for both types of images, but the initial problem remains: absolute positioning causes page flow complications. Absolute positioning is needed to stack images on each other, but it does this because removes an element from the object flow of the page, and needs a container with a set size in order to properly space things out. The figcaptions have to have a padding amount equal to the font size, plus the image size, plus a bit more to space properly below the slideshows, and that's the only thing keeping the layout together.

My plan is to work another method of doing the slideshow (and trust me, there's tons of ways). I want to make the documentation as accessible as possible to everyone, and the number of hacks currently necessary even in the unmodified layout just makes things complicated. It's not good for development if someone has to look at multiple files in different languages just to figure out how a single element works.

So in the mean time I've removed the image scaling but kept the centering. Considering how many snags I've run into, and now doing a redo for review purposes, I'm going to try and keep everything as similar as possible in terms of HTML.
- The centering is done behind the scenes using preexisting CSS styling so it won't get in the way.
- Closing the tags isn't just saved until my last step, I'm going to leave it until you want to push a release. It'll literally be THE last thing that's done, period.
- The slideshow may get updated, but that shouldn't cause dramatic conflicts. Even so, it's going to be left until after content correction anyway.

Also, I would still appreciate it if you would let me know what branches are going to be heading into the release, or at least the currently suspected ones. That way I can pull their docs and add any changes in.

**Comment by Utumno** [comment]
> * Closing the tags isn't just saved until my last step, I'm going to leave it until you want to push a release.

Exactly my thoughts - so we leave this for next major release and will merge `grnbeef` branch  you are ready (the sooner the better)

> what branches are going to be heading into the release

We are in the process of merging various branches and more pop up - but since we won't have the closing tags practically rewriting the readmes there won't be impossible conflicts. So don't care about those just let me know when your grnbeef branch is ready so I can merge (solving the little conflicts and removing the merge commits)

Thanks!

**Comment by FelesNoctis** [comment]
Can do. Going to do another run over my previous commits and double-check a couple of things from some of the nightlies that I noticed had mild differences.

Do you know off the top of your head if anyone else has done screenshot work? I need to do some new ones for a couple of areas that weren't covered, and should also update for some of the changes that were made. No point in doubling up on screenshots. And if that is entirely on me, what branch(s) should I grab so I can get the UI updates?

**Comment by FelesNoctis** [comment]
Oh, and it just occurred to me, we probably need to write a whole new section on ESLify. Possibly in General and in Advanced, depending on how much detail we need.

**Comment by Utumno** [comment]
@FelesNoctis - is the BBQ ready? :P

Don't bother carrying the changes from utumno-wip will do that - just finalize the cooking

Thanks!

**Comment by FelesNoctis** [comment]
@Utumno Yeah, I'll let you add the toppings :P

I'm planning on doing as much as I can of General and Technical as I can. However, removing the `-rebased` branch has slowed things down, I was using that for comparisons for the content changes.

**Comment by Utumno** [comment]
Ooops sorry - but I was sure you had a local copy :(

**Comment by FelesNoctis** [comment]
I do, but you know me, I have a hard time with git :P I was basically just using Github's interface to compare what changed where and going through the commits in order. After the drastic 90% change, it's really easy to catch the content changes.

**Comment by FelesNoctis** [comment]
Can we recover it temporary? I was looking at reflog and checking out by SHA, but I'm not sure that does what I need and I'm not entirely comfortable with it anyway.

**Comment by FelesNoctis** [comment]
Hold on, I think I got it. I can see the commits on one of my copies, and I believe this goes up to the last before the swap to the simplified version. As long as I don't prune (I think) this should be fine. Gitkraken makes the changes more obvious (similar to Github) than Gitk, so I should be able to work with this.

**Comment by FelesNoctis** [comment]
There was no other way.

**Comment by Utumno** [comment]
Hey @FelesNoctis is this done you think?

**Comment by Utumno** [comment]
Hahaha  Mr.Utumno will soon review and merge with help from @wrye-bash/bashers

**Comment by FelesNoctis** [comment]
There's always the post-credits scene, but that'll end up in another branch for those who care to stick around. :P

But yes, that should be everything you originally asked for, plus some minor corrections and additions. Fixed missing content menu options, updated tables to match the context menu images I added, style updates to make it a bit easier to read, etc.

**Comment by FelesNoctis** [comment]
Oops, hold on just a moment. I've been running them through validators and there are a few things I forgot about. Errors I believe were corrected before from the formatting sweep originally.

**Comment by FelesNoctis** [comment]
Alright, NOW it should be all good to go.

Oh, and this was tested at 1024x768 as well as 1080x1920 and 1920x1080. Should be fine across the board. :P

**Comment by Utumno** [comment]
Thanks @FelesNoctis :)

@wrye-bash/bashers if anyone can review this before I get there appreciated!

**Comment by FelesNoctis** [comment]
@Utumno 
What is `feles-docedit-postcred-dev`?
`-postcred` is an experimental build where I'm adding features. **I do not, for any reason, recommend preparing it for merge for a release at this time.** The styling and scripting are going to be changing wildly from one version to the next.

If you're concerned about `-grndbeef`'s cherry-picks, those only contain the readme edits made between the time that I started my documentation work (roughly) and now, **ONLY** from along the path of branches that were merged into `dev` during that time.

I do recommend however grabbing the last commit in `-grndbeef` however, as the javascript file name change will be essential to any UI changes coming from `-postcred` in the future.

**Comment by FelesNoctis** [comment]
I'm reopening this issue, and have updated the first post with current status information.

**Comment by Utumno** [comment]
Ok I see - the correct procedure is not to cherry pick the dev content in your branch but instead rebase your branch on dev - if you cherry pick you create essentially the same content that is on dev (in a mangled up state) on your branch. See: https://github.com/wrye-bash/wrye-bash/wiki/%5Bgithub%5D-Branching-and-merging-to-dev-using-rebase

But anyway it's ok - I proceeded to rebase your branch on dev myself https://github.com/wrye-bash/wrye-bash/commits/feles-docedit-grndbeef-dev

Please review there is all that's needed in there - I will squash a bit more and merge

If the `postcred` is not to be merged then it should be removed from the repository. We are trying to reduce branch count here (the hydra as I call it). If you want to share your wip stuff you should redo your fork and push there (see wiki above)

Anyway I rebased your experimental branch on dev too so you have a reference if you try rebasing yourself. But in the next few days please fetch those and delete them from here -apart from `feles-docedit-grndbeef-dev` where you should add any last minute fixes so we (finally) merge

Thanks :)

**Comment by FelesNoctis** [comment]
@Utumno 
So two things:
1. I had no idea you could rebase onto the same location the branch is already sitting. Thank you for letting me know, though my brain's trying to wrap itself around that one. :P 
2. I want to clarify ahead of time that this is not aimed at you. I looked at all the options and tossed out the ones I'd been told were a bad idea, and researched my best and several detailed resources had pictures so I could learn like I'm 5, and cherry-pick seemed like exactly what I needed since I only wanted doc changes specifically and nothing else, because that's the whole point of my branch so I used that, and _**why the hell does git have some many ways to do basically the same thing?**_

*cough*

Anyway. I'll take a look at -grndbeef-dev to make sure it's accurate. It should be, you know how to handle this stuff better than I do. :P If you want me to move the experimental branch to my fork I can (that's why I have my local set up the way I do, so I can switch repos if I need to!), but I should be able to make a commit today that puts it in a stable state for adding to dev.

**Background on that:** the sticky headers I implemented require a bunch of added boxes to make them function correctly (they need to have containers that tell them where to begin and end), but those boxes and the spacers added for padding are annoying even for ME to set up, and I'm the one that made the system. So what I'm planning on doing today is implementing into the sticky javascript that's already there some DOM manipulation to add the tags by script. That way we're back to you guys making whatever changes you want easily, and the JS just handles adding all the boxes and spacers on page render. Also due to this reason I'll likely be removing the CSS redundancy for the sticky headers so it defaults back to normal layout.

**Comment by Utumno** [comment]
Ok thanks ! Yes please review the grndbeef-dev - pay attention to the sections that were edited in also on dev and so when I rebased there were conflicts there. Add fixups commits if any on `grndbeef-dev`, and add also the commits you need from the postcred. Move all other branches to your fork and delete them from here

As for git - cherry picking is adding to your branch but rebase is just cutting all your commits and rebasing them on the new HEAD of dev branch - that's what you need here.

Do use the tree - `gitk --all &` - this is the only way to grok git

Thanks!

**Comment by FelesNoctis** [comment]
"Grok"
Oh jeeze, this is getting too nerdy for me :P 

Had to do some errands, but I'm taking a look now. It looks like your rebase did miss some of the newer changes, so I'm working on that.

**Comment by FelesNoctis** [comment]
Didn't realize I could delete branches until just now! `-grndbeef` is gone, dev as far as I can tell is up to speed. Now I need to make sure `-postcred-dev` is in good shape before I make the javascript changes.

**Comment by Utumno** [comment]
Excellent @FelesNoctis - you are getting there with git I'd say too - do use gitk (or git gui > Repository > visualizer all branch history).

I will rewrite the history a bit to squash fixups and merge - keep working on -postcred-dev - if you can move it to your fork cause I want to clean up this repo a bit - I have countless local branches and it's difficult for me to navigate the tree, especially the latest part

_Once I merge on dev_ you should rebase your branch issuing `git rebase --interactive --onto dev 0cb2aae2c64 feles-docedit-postcred-dev` - this tells git to cherry pick all commits that are on `feles-docedit-postcred-dev` after 0cb2aae2c64 which is the HEAD of the feles-docedit-grndbeef-dev branch, which will have been absorbed to dev

**Comment by FelesNoctis** [comment]
As of last night, `feles-docedit-postcred-dev` is actually the preferred branch. I stripped out the sticky headers due to complications. Specifically, I was trying to make things easy for editing by using the JS to add the required styling boxes, but that isn't going to work. The alternatives are either cluttering the page with manually-added tags, or removing it altogether. I opted for the latter.

In any case, both `-postcred` and `-grndbeef-dec` can be tossed. `grndbeef-dev` can probably wait until you do merge, and then I can rebase `-postcred-dev` on top of that instead. Either way.

The next commit has a slide menu. I may throw it on my fork first so that you can take a look, but I think it's going to be a much better option than having the TOC at the top. Another alternative is to just add a "back to top" link at every header. *shrug*

**Comment by FelesNoctis** [comment]
@Utumno 
https://felesnoctis.github.io/wrye-bash-docedit/

Current progress on heavier restyling. Don't bother on mobile. It technically works, but it ain't pretty. Obviously it's extremely low priority but I have plans for that as well.

**Comment by FelesNoctis** [comment]
@Utumno 
From Discord:

> Yes it's Utumno and yes I am on Discord - @FelesNoctis I finally (finally) got round to rebasing feles-docedit-postcred-dev on current dev: https://github.com/wrye-bash/wrye-bash/tree/feles-docedit-postcred-dev-final . I spent around five hours doing so (that I could not afford till now). I would like you to review that so I can merge. Turns out that massively editing the readme while developing was not a good idea (as there were constantly conflicts) so a readme overhaul should take place in parallel with the main code repository. But I would like to merge your changes (not th eleast reasion being a token of respect for what you have been through with git :stuck_out_tongue: ).So please have a look at that final branch and let me know if everything is ok (preferably on the relevant issue on the main repo). I rearanged/squashed etc but kept your commit messages - just wrapped to 72 chars. The content of that branch should be identical to a1bf3f9a601f8c409ef211639920059a25e496f2 apart from some tiny whitespace changes and the edits by @Infernio from the current dev branch.

*deep breath*

If you look just above this post, you'll notice that I've been working on updating the documentation on a completely separate fork, which has existed for 3 months now. Said fork is currently in a state where, with a few minor visual adjustments and updating with the current documentation changes, could be used as the documentation for the next release. I'd been keeping it updated until that point. If it took you 5 hours to merge the `postcred-dev` changes into a dev-ready state, then I'm sorry, but you probably wasted 5 hours. That version has a lot of leftover trash in it that's been long since removed.

The biggest issue here that I see is the lack of communication. You've really gotta try to communicate with us more. If you'd asked, I would have told you that the privately developed version is in a much better position than the one you just worked on.

As it stands now, if you're reading to push the documentation into dev, I'm going to finish up the minor things I'd left stagnant in my fork, compare against your `-final` to make sure I have the documentation content the way you want it, and then overwrite on that branch.

And I really am sorry for calling you out in public like this, but it's the only reliable method I seem to have to contact you at the moment. I'm more than willing to do the grunt work for you if you asked.

**Comment by Utumno** [comment]
Hey sorry, @FelesNoctis it's not lack of communication rather lack of time (plus html/css/javascript is not really my area of expertise). If you want to proceed and update your fork with current changes feel free to do so (if you compare feles-docedit-postcred-dev there are mainly some changes by Infernio and the gimpshop commands). I had seen the fork but I thought it was a complete rewrite so to keep things simple I thought I could just rebase your branch and include those changes

If the community finds the readme pretty enough and all the latest changes are carried along (you can use `$ gitk --all -- Mopy/Docs/ &` on the git Bash to see those) then it's ok by me

Thanks!

**Comment by FelesNoctis** [comment]
Don't worry, I'll make sure all the doc changes are up to speed.

And yes, it was a much larger undertaking, but it's been (mostly) ready to go since I made the post on March 18th. I was hoping you'd take a look and respond before I did much more with it, since I didn't want to throw on the final polish until I had opinions. I have everybody but yours now, other than "ok by me". I'll take that as an endorsement I suppose :P

**Comment by FelesNoctis** [comment]
And I do feel terrible about the confusion there, I'd thought I'd already let everyone know the situation. That's time I wish you could get back.

**Comment by Utumno** [comment]
> I'll take that as an endorsement I suppose :P

Sure

> And I do feel terrible about the confusion there, I'd thought I'd already let everyone know the situation. That's time I wish you could get back.

Haha no worries

> The bigger overhaul hasn't been started yet, as it'll look absolutely nothing alike and merging would be impossible. If and when the time comes it would be a full replacement.

Yep I guessed so - well - 308 I guess

**Comment by FelesNoctis** [comment]
Alright, everything's merged together on my fork.
https://felesnoctis.github.io/wrye-bash-docedit/

There's some minor changes I want to make before I merge it into the main branch. Menu functionality so it doesn't close the category when you click a subsection, updating links to the Beth forums to go to the current/correct location, alphabetizing the General/applications sections, pull out the javascript toggle link, etc.

Tomorrow's pretty open for me schedule wise, particularly since I'm dealing with a flu, so hopefully I'll be able to wrap everything up and drop it over here soon.

**Comment by Utumno** [comment]
Great!!

I would probably prefer to just add a single commit to the main dev branch - can you point me to the branch you are working on?

**Comment by Utumno** [comment]
Oh found it : https://github.com/FelesNoctis/wrye-bash-docedit/commits/dev

Actually, it does have a nice commit history - so you took my final branch and build on top of it. Great! I am loooking forward to the final corrections you referred to. Be sure to ask people to have a look to the final result - the more testers the better.

Thanks!

**Comment by Utumno** [comment]
Apart from the last commit, that's not needed probably - some of it it's because you actually forked Wrye-code-collection instead of this repo. I would ask you to fork this one but after this merge is over

**Comment by FelesNoctis** [comment]
It's actually the other way around. This was forked off of Collection back when we had the "incident" and I'd lost my push permissions for a day or so.

I did some cleanup of my fork first because I had something like six branches, most of which were garbage. What was left was my old `-postcred` branch and my master, and I duplicated the former into a new dev. 

"Minor changes" and the three cleanup passes are the recent work, and then the merge is folding a duplicate of your `-final` branch into my dev. Once I was fairly confident everything was the way it should be I got rid of the old branch.

TL;DR I still hate git but I'm learning :P

So yeah. Tried to keep the commits clear, clean, and consistent so you can see what's going on. Same goes for the next changes, and I'll let you know when it's good for another check. And afterwards yeah, I was planning on deleting and reforking from here instead.

**Comment by FelesNoctis** [comment]
Looking back at the history, I think we're seeing the duplicate of your `-final`'s history. If you go back to late march you can see where my changes broke off from the branch here.

Apparently I didn't squash the commits.

I still stick to my story that I'm learning!

**Comment by FelesNoctis** [comment]
Alright. Not QUITE done (there's a few more fiddly bits to take care of like CSS sheet consistency), but I should head to bed. I'll pick it up tomorrow. ALMOST there!

https://github.com/FelesNoctis/wrye-bash-docedit/commits/dev

**Comment by FelesNoctis** [comment]
@Utumno 
Holy hell that was a lot more work than I was expecting it to be! It should be ready to go now, and the master's also updated for full preview.

https://github.com/FelesNoctis/wrye-bash-docedit/commits/dev
https://felesnoctis.github.io/wrye-bash-docedit/

**Comment by FelesNoctis** [comment]
I just realized, I don't know why we don't have an index file in the actual documentation folder itself. I'll deal with that tomorrow since I need to be up in about... 5 hours, but that's a change that takes a minute at most.

**Comment by Utumno** [comment]
Hey I  start to rebase (btw your woes with git stem from the fact that you do merges instead of rebases). So I will rebase your recent changes over https://github.com/wrye-bash/wrye-bash/tree/feles-docedit-postcred-dev-final - the final result will be all your changes positioned over dev and ready for merge (!). I have some questions though:  in https://github.com/FelesNoctis/wrye-bash-docedit/commit/811dee1d285a9bc88b728689e41ff75831c6a3aa you replace a lot of spaces with tabs - is this needed?

**Comment by FelesNoctis** [comment]
On the commit history:
I'd actually attempted to rebase twice, but it kept starting me all the way back at the very  beginning of my changes, and then would run me through the history itself twice (once for the main repo, once for mine, since I forked off of Collection?). I got so frustrated with the repeat garbage it threw at me that had long since been replaced that I finally just did a merge and did the diffs all in one go. If anything I'd say my woes came from rebasing, and probably poor choices early on in my work :P

On the whitespace:
Nah, not necessary at all. That might explain the numerous "these are conflicts but for some reason their content is identical" issues I had. My editor is set to use tabs by default so that'd definitely explain it. Sorry about the mess then, I suppose! Though if I remember correctly that's part of the reason I wanted to reformat in the first place: it was mixed in a lot of places already. 

I did check it one more time before I headed to bed last night, and the only changes from `-final` here were my additions in Docs and images as far as I could tell (which would make sense), so you should be able to tell it to accept everything.

**Comment by FelesNoctis** [comment]
To clarify the whitespace, if it's the seemingly identical ones in the HTML files, those are the ones I ran into that had me scratching my head. If it's in the CSS or JS, that's likely just due to my experimenting with settings and the editor formatting with its defaults.

Also I just realized I forgot to re-add the compatibility code for various browsers. Go ahead and finish if you're still working on it, I'll just add that and the index file here directly. This should be the end of that repo for this version of the docs anyway.

**Comment by FelesNoctis** [comment]
![Screenshot_20190607-091532_Firefox](https://user-images.githubusercontent.com/16215725/59118532-42cbb900-8905-11e9-8a9f-e3a281939ad6.jpg)
Ugh. Since I'm the only one really who touches the stylesheets, would you hate me if I at least did a formatting pass over those? That whitespace inconsistency is going to irk me :P I think it was all spaces to begin with if the older page-specific styles are any indication, 2 spaces per level, so I may do that as part of my final touches.

**Comment by Utumno** [comment]
Ok I have rebased - took all day cause I removed the tabs and then I had a gazillion of conflicts.

https://github.com/wrye-bash/wrye-bash/tree/feles-docedit-postcred-dev-final

The diffs from your commit:

```
From 04dec30db4bbc4982a5b6e46bc699e7350bbcb2b
     Tuning Pass, Section 7: Index + Opt + Bugfix

To   a127352fedfcffce8a816dc59aa687692d8765a3
     Tuning Pass, Section 6: CSS (Style Tweaks)
```

only differ in a couple newlines and that `<div class="clear"></div>` which I ommited in some stage readd it back please if needed:

![feles-diff](https://user-images.githubusercontent.com/1365079/59127786-1e0d2b00-8971-11e9-8448-a1ad100c9ea4.jpg)

Note that if I do not look into white space - I trimmed whitespace from the end of the lines

Please apply any new changes on top of this branch (you can copy it over to your repo by issuing

```
$ git fetch git@github.com:wrye-bash/wrye-bash.git feles-docedit-postcred-dev-final
$ git checkout -b feles-docedit-postcred-dev-final FETCH_HEAD
```
)

Let's finish with this please - forgot to mention it does look good :P 

Oh - and what is that `index.html` file? Does it belong here or to the readme repo?

**Comment by FelesNoctis** [comment]
The `index.html` at the root level should go to the readme repo when this is all settled and done. I'm going to make a copy in the `Docs` folder itself and do some minor adjustments so users have a proper start point (and maybe we can link that in the program itself).

I'll do a quick run over it and make the last of the changes. ALMOST THERE, FINALLY! :P

**Comment by FelesNoctis** [comment]
![accessrights](https://i.imgur.com/zp7lVH6.png)

Sir, not sure that's going to work :P

**Comment by Utumno** [comment]
$ git fetch https://github.com/wrye-bash/wrye-bash.git feles-docedit-postcred-dev-final

**Comment by FelesNoctis** [comment]
Oh it's ok, I've had dual remotes set up on this local for a while now. I'm already working on going through the changes. :P

The extra white-space in the HTMLs was meant as a divider between sections. Separate head from body and all that. I can leave it out if you like, I just figured it made things a bit easier to read. I've never liked major nodes crammed up against each other, personal preference.

**Comment by Utumno** [comment]
On the other hand reducing newlines helps being able to navigate more... I leave it to you. 

Is `<div class="clear"></div>` needed in the Technical? If yes please put it back in

**Comment by FelesNoctis** [comment]
Yes, and I already re-added it. It provides a proper break between the Contents lists and the header below in non-javascript implementation for older browsers. Though I may swap it to `clearf`, not sure why the old version is still in there.

**Comment by FelesNoctis** [comment]
![keepcalmdontpanic](https://user-images.githubusercontent.com/16215725/59143686-3e91b100-8982-11e9-903d-1d6528b1b566.jpg)

I think we made it, Mister Utumno.

**Comment by Utumno** [comment]
Great great - ehmmm - where is da code? :P

**Comment by FelesNoctis** [comment]
Exactly where you wanted me to put it? :P

https://github.com/wrye-bash/wrye-bash/commits/feles-docedit-postcred-dev-final

**Comment by Utumno** [comment]
Great fantastic - now do we need the two index files here or should I move them to the other repo (the readme one). I do not think they make sense here

Should I keep both?

**Comment by FelesNoctis** [comment]
`. \index.html`
Readme repo specific, not needed in the distribution package. 

`Mopy\Docs\index.html`
Intended to be included with the distribution. Is basically the same but with localized links. Wrye Bash could also point to this as the start of the help file, if we wanted to make that change.

**Comment by Utumno** [comment]
Thanks - will turn to it. I am thinbking of carying the history of your commits in the readme repo (which means I shouldn't have edited them so much cause clean history belongs to here) and add here a commit with all the changes.

Not sure if we should redirect to the index makes sense to show the General readme first - or not? What do people say?

**Comment by FelesNoctis** [comment]
Yeah, it's just there as an option if we decide it's useful. Given that I added a navbar to the bottom of the pages, the only things we'd be lacking without the index are links to the online readme repos, which isn't a huge loss :P

**Comment by FelesNoctis** [comment]
Oh, as for the history, I honestly wouldn't stress about it if I was in your position. This is likely the last of any big changes to this version of the readme, with the complete rebuild planned for 308/309. A little bit of dirty history won't hurt anyone, and saves you some sanity.

Also, I think we still need to add a FOMOD-related section, and update the installer menu screenshots for that

**Comment by Utumno** [comment]
Alea jacta est @wrye-bash/bashers and @FelesNoctis (thanks!). Rewritten docs are on dev. Although I am really satisfied with end result, turned out that editing the readmes and keeping the rewrite in par with edits by @Infernio and me, plus my OCD for a clear history has cost me a _lot_ of time (as in _a lot_) that should go to the many outstanding issues - especially the wx/p3 update. Sooo, let's keep rewrites of docs out of the main repo - feel free to work on it on forks etc but next time let's just overwrite with the final result (care must be taken that fixups from main repo should be ported)

Back to business - I will proceed to rebase the nightly branch and cherry pick some of it on dev

Happy Bashing!
